streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -1088,4 +1090,68 @@ public void shouldCreateStateDirectoryForStatefulTopology() {
         final TaskId taskId = new TaskId(0, 0);
         assertTrue(new File(appDir, taskId.toString()).exists());
     }
+
+    @Test
+    public void shouldReturnIteratorForMultipleRecords() {
+        testDriver = new TopologyTestDriver(setupSourceSinkTopology(), config);
+
+        testDriver.pipeInput(consumerRecord1);
+        final byte[] key2 = "key2".getBytes();
+        final byte[] value2 = "value2".getBytes();
+        final long timestamp2 = 42L;
+        final ConsumerRecord<byte[], byte[]> consumerRecord2 = consumerRecordFactory.create(SOURCE_TOPIC_1, key2, value2, headers, timestamp2);
+        testDriver.pipeInput(consumerRecord2);
+
+        final Iterator<ProducerRecord<byte[], byte[]>> output = testDriver.iterateOutput(SINK_TOPIC_1).iterator();
+
+        final ProducerRecord outputRecord1 = output.next();
+
+        assertEquals(key1, outputRecord1.key());
+        assertEquals(value1, outputRecord1.value());
+        assertEquals(SINK_TOPIC_1, outputRecord1.topic());
+
+        final ProducerRecord outputRecord2 = output.next();
+
+        assertEquals(key2, outputRecord2.key());
+        assertEquals(value2, outputRecord2.value());
+        assertEquals(SINK_TOPIC_1, outputRecord2.topic());
+        assertFalse(output.hasNext());
+    }
+
+    @Test
+    public void shouldReturnIteratorForMultipleRecordsWithSerDes() {
+        testDriver = new TopologyTestDriver(setupSourceSinkTopology(), config);
+
+        testDriver.pipeInput(recordFactory.create(SOURCE_TOPIC_1, "Key1", 12345L));
+        testDriver.pipeInput(recordFactory.create(SOURCE_TOPIC_1, "Key2", 6789L));
```

Comment:
Could you please extract the values `"Key1"`, `12345L`, `"Key2"`, and `"6789"` into variables and use the variables in the checks below? I am sorry, I missed that in my previous review.

Reply:
Done